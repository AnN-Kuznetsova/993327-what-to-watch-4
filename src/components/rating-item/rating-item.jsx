import PropTypes from "prop-types";
import React from "react";


export const RatingItem = (props) => {
  const {
    id,
    onClick,
  } = props;

  const handleRatingChange = (event) => {
    event.target.checked = true;
    onClick();
  };

  return (
    <React.Fragment>
      <input
        className="rating__input"
        id={`star-${id}`}
        type="radio"
        name="rating"
        value={id}
        onChange={handleRatingChange} />
      <label className="rating__label" htmlFor={`star-${id}`}>Rating {id}</label>
    </React.Fragment>
  );
};


RatingItem.propTypes = {
  id: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};
