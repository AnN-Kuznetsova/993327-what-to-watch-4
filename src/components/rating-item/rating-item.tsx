import PropTypes from "prop-types";
import React from "react";


export const RatingItem = (props) => {
  const {
    id,
    onChange,
    rating,
  } = props;

  return (
    <React.Fragment>
      <input
        className="rating__input"
        id={`star-${id}`}
        type="radio"
        name="rating"
        value={id}
        checked={id === rating}
        onChange={onChange}
      />
      <label className="rating__label" htmlFor={`star-${id}`}>Rating {id}</label>
    </React.Fragment>
  );
};


RatingItem.propTypes = {
  id: PropTypes.number.isRequired,
  rating: PropTypes.number,
  onChange: PropTypes.func.isRequired,
};
