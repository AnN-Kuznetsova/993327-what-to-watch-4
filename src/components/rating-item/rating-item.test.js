import * as React from "react";
import renderer from "react-test-renderer";

import {RatingItem} from "./rating-item";


describe(`Render RatingItem`, () => {
  it(`RatingItem should match with snapshot`, () => {
    const props = {
      id: 1,
      rating: 2,
      onChange: () => {},
    };

    const ratingItemSnapshot = renderer.create(
        <RatingItem {...props} />
    ).toJSON();

    expect(ratingItemSnapshot).toMatchSnapshot();
  });


  it(`RatingItem should match with snapshot when it is checked`, () => {
    const props = {
      id: 1,
      rating: 1,
      onChange: () => {},
    };

    const ratingItemSnapshot = renderer.create(
        <RatingItem {...props} />
    ).toJSON();

    expect(ratingItemSnapshot).toMatchSnapshot();
  });
});
