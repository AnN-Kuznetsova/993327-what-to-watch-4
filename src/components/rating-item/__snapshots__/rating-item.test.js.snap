// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RatingItem RatingItem should match with snapshot 1`] = `
Array [
  <input
    checked={false}
    className="rating__input"
    id="star-1"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={1}
  />,
  <label
    className="rating__label"
    htmlFor="star-1"
  >
    Rating 
    1
  </label>,
]
`;

exports[`Render RatingItem RatingItem should match with snapshot when it is checked 1`] = `
Array [
  <input
    checked={true}
    className="rating__input"
    id="star-1"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={1}
  />,
  <label
    className="rating__label"
    htmlFor="star-1"
  >
    Rating 
    1
  </label>,
]
`;
